builder/dockerfile/containerbackend.go

```diff
@@ -0,0 +1,156 @@
+package dockerfile
```

Comment:
Builder doesn't need to depend on one central dependency(builder adaptor). It can just talk with "image API" and "container API" directly. In #32904 the "container API" is `Exec()` and `Debug()`. There is nothing builder specific in these methods so it is wrong to call it "builder *". That being said, there is no requirement that "container API" needs to be `daemon.Daemon` instance. I'd be happy if you can move it anywhere outside of `builder/` dir.